aptos-move/move-examples/fungible_asset/sources/managed_fungible_asset.move

```diff
@@ -50,116 +64,286 @@ module fungible_asset_extension::managed_fungible_asset {
+        let mint_ref = if (*vector::borrow(&ref_flags, 0)) {
+            option::some(fungible_asset::generate_mint_ref(constructor_ref))
+        } else {
+            option::none()
+        };
+        let transfer_ref = if (*vector::borrow(&ref_flags, 1)) {
+            option::some(fungible_asset::generate_transfer_ref(constructor_ref))
+        } else {
+            option::none()
+        };
+        let burn_ref = if (*vector::borrow(&ref_flags, 2)) {
+            option::some(fungible_asset::generate_burn_ref(constructor_ref))
+        } else {
+            option::none()
+        };
         let metadata_object_signer = object::generate_signer(constructor_ref);
         move_to(
             &metadata_object_signer,
-            ManagedFungibleAsset { mint_ref, transfer_ref, burn_ref }
+            ManagingRefs { mint_ref, transfer_ref, burn_ref }
         )
     }
 
-    /// Mint as the owner of metadata object.
+    /// Mint as the owner of metadata object to the primary fungible stores of the accounts with amounts of FAs.
+    public entry fun mint_to_primary_stores(
+        admin: &signer,
+        asset: Object<Metadata>,
+        to: vector<address>,
+        amounts: vector<u64>
+    ) acquires ManagingRefs {
+        let receiver_primary_stores = vector::map(
+            to,
+            |addr| primary_fungible_store::ensure_primary_store_exists(addr, asset)
+        );
+        mint(admin, asset, receiver_primary_stores, amounts);
+    }
+
+
+    /// Mint as the owner of metadata object to multiple fungible stores with amounts of FAs.
     public entry fun mint(
         admin: &signer,
-        metadata: Object<Metadata>,
-        amount: u64,
-        to: address
-    ) acquires ManagedFungibleAsset {
-        let managed_fungible_asset = authorized_borrow_refs(admin, metadata);
-        let to_wallet = primary_fungible_store::ensure_primary_store_exists(to, metadata);
-        let fa = fungible_asset::mint(&managed_fungible_asset.mint_ref, amount);
-        fungible_asset::deposit_with_ref(&managed_fungible_asset.transfer_ref, to_wallet, fa);
+        asset: Object<Metadata>,
+        stores: vector<Object<FungibleStore>>,
+        amounts: vector<u64>,
+    ) acquires ManagingRefs {
+        let length = vector::length(&stores);
+        assert!(length == vector::length(&amounts), error::invalid_argument(ERR_VECTORS_LENGTH_MISMATCH));
+        let mint_ref = authorized_borrow_mint_ref(admin, asset);
+        let i = 0;
+        while (i < length) {
+            fungible_asset::mint_to(mint_ref, *vector::borrow(&stores, i), *vector::borrow(&amounts, i));
+            i = i + 1;
+        }
     }
 
-    /// Transfer as the owner of metadata object ignoring `frozen` field.
+    /// Transfer as the owner of metadata object ignoring `frozen` field from primary stores to primary stores of
+    /// accounts.
+    public entry fun transfer_between_primary_stores(
```

Comment:
Hmm I think this is making this code a lot more complex. Is there may be a better way to organize things?

Reply:
This is another PR I stacked on. It will be landed first.